src/components/Chip/Chip.tsx

```diff
@@ -1,19 +1,32 @@
 import React from "react";
 import PropTypes from "prop-types";
 import { highlightSubString } from "../../utils";
+import type { KeyboardEvent, MouseEvent } from "react";
+
+export type Props = {
+  lead?: string;
+  onClick: (
+    event: MouseEvent<HTMLDivElement> | { lead: string; cloud: string }
```

Comment:
This event type seems weird. The lead/cloud thing seems project specific (I guess it's just a leftover from JAAS). And from implementation it seems that onClick works differently when clicking (returns event) and using keyboard (returns arbitrary object).

Probably out of scope for this migration PR, but we should have a separate issue to address this.

Reply:
I've filed an issue https://github.com/canonical-web-and-design/react-components/issues/507.